contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,853 @@
+    spec:
+      containers:
+      - name: nginx
+        image: gcr.io/google_containers/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
+        volumeMounts:
+        - name: www
+          mountPath: /usr/share/nginx/html
+  volumeClaimTemplates:
+  - metadata:
+      name: www
+      annotations:
+        volume.alpha.kubernetes.io/storage-class: anything
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 1Gi
+```
+
+Users can use the following command to create the StatefulSet.
+
+```shell
+kubectl create -f web.yaml
+```
+
+The only difference between the proposed and current implementation is that 
+the proposed implementation will initialize the StatefulSet's revision history 
+upon initial creation.
+
+### Rolling out an Update
+Users can create a rolling update using `kubectl apply`. If a user creates a 
+StatefulSet [as above](#initial-deployment), the user can trigger a rolling 
+update by updating image (as in the manifest as below).
+
+```yaml
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: web
+spec:
+  serviceName: "nginx"
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: gcr.io/google_containers/nginx-slim:0.9
+        ports:
+        - containerPort: 80
+          name: web
+        volumeMounts:
+        - name: www
+          mountPath: /usr/share/nginx/html
+  volumeClaimTemplates:
+  - metadata:
+      name: www
+      annotations:
+        volume.alpha.kubernetes.io/storage-class: anything
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 1Gi
+```
+
+
+Users can use the following command to trigger a rolling update.
+
+```shell
+kubectl apply -f web.yaml
+```
+
+### Canaries
+Users can create a canary using `kubectl apply`. The only difference between a
+ [rolling update](#rolling-out-an-update) and a canary is that the 
+ `.Spec.GenerationPartition` is set to `.Spec.Replicas - 1`.
+ 
+```yaml
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: web
+spec:
+  serviceName: "nginx"
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: gcr.io/google_containers/nginx-slim:0.9
+        ports:
+        - containerPort: 80
+          name: web
+        volumeMounts:
+        - name: www
+          mountPath: /usr/share/nginx/html
+     generationParition: 2
+  volumeClaimTemplates:
+  - metadata:
+      name: www
+      annotations:
+        volume.alpha.kubernetes.io/storage-class: anything
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 1Gi
+```
+
+Users can also simultaneously scale up and add a canary. This reduces risk 
+for some deployment scenarios by adding additional capacity for the canary. 
```

Comment:
You get autoscaled during an update? You would get the new version. I think you'd want to disable autoscaling during updates, but, if you didn't, auto scaling to the updated version is reasonable.